cognite/client/_api/diagrams.py

```diff
@@ -230,14 +237,31 @@ def detect(
                     ]
                 }
                 </code>
+
+            To use beta configuration options you can use a dictionary or `DiagramDetectConfig` object for convenience:
+
+                >>> from cognite.client.data_classes.contextualization import ConnectionFlags, DiagramDetectConfig
+                >>> config = DiagramDetectConfig(
+                ...     remove_leading_zeros=True,
+                ...     connection_flags=ConnectionFlags(
+                ...         no_text_inbetween=True,
+                ...         natural_reading_order=True,
+                ...     )
+                ... )
+                >>> job = client.diagrams.detect(entities=[{"name": "A1"}], file_id=123, config=config)
+
+            Check the documentations for `DiagramDetectConfig` for more information on the available options.
         """
         items = self._process_file_ids(file_ids, file_external_ids, file_references)
         entities = [
             entity.dump(camel_case=True) if isinstance(entity, CogniteResource) else entity for entity in entities
         ]
         beta_parameters = {}
         if pattern_mode is not None or configuration is not None:
-            beta_parameters = dict(pattern_mode=pattern_mode, configuration=configuration)
+            config = configuration.dump() if isinstance(configuration, DiagramDetectConfig) else configuration
+            beta_parameters = dict(pattern_mode=pattern_mode, configuration=config)
+            self._detect_beta_params_warning.warn()
+            self._api_subversion = "beta"
```

Comment:
@haakonvt does this look correct? I saw the datapoints endpoint appends `"-beta"` to subversion but setting subversion to `"beta"` seems to work for `digrams.detect`.

Reply:
The API versioning guide is very clear on what format it should be:
https://cognitedata.atlassian.net/wiki/spaces/PD/pages/3842900086/CDF+API+versioning

Reply:
I think beta is equivalent to <latest>-beta, but better safe than sorry maybe

Reply:
Oh, thanks for sharing! I wasn't aware of that doc.

Will change to append '-beta'.

Reply:
Hmm, I'm not sure why the integration test is saying the api subversion is bad now 🤔

Reply:
@olacognite `Invalid version in cdf-version header`... 😉

Reply:
(Just double-check that it is in fact sending `"20230101-beta"`)